smoke test pipeline hook

ckairen opened the pull request.

ckairen committed:
smoke test pipeline hook

ckairen committed:
smoke test pipeline hook

jsquire reviewed (commented):
Can you help me understand what the structure of the smoke test project will look like?  I see the available packages being output by `BuildSmokeTestProject.yml` but I don't see where the `.csproj` is generated.

jsquire committed:
[Smoke Tests] Restructure for Conflict Validation

The focus of these changes is to restructure and update the smoke tests
to validate that multiple Azure SDK packages can be used in the same application
without encountering conflicts for dependency versions.

This validation is focused only on ensuring that the assemblies do not
conflict and makes no attempt to validate scenarios from any specific
library.  Verifying library functionality is out of scope, and assumed to
be the responsibility of the library's test suite.

jsquire committed:
[Smoke Tests] Restructure for Conflict Validation

The focus of these changes is to restructure and update the smoke tests
to validate that multiple Azure SDK packages can be used in the same application
without encountering conflicts for dependency versions.

This validation is focused only on ensuring that the assemblies do not
conflict and makes no attempt to validate scenarios from any specific
library.  Verifying library functionality is out of scope, and assumed to
be the responsibility of the library's test suite.

ckairen commented on eng/scripts/smoketest/Save-Package-Properties.ps1:
```diff
@@ -0,0 +1,122 @@
+  {
+    $outputObject = $incomingPackageSpec
+  }
+  
+
```
https://gist.github.com/ckairen/f638231d6d146fef7b73f35d6ca7f7e0#smoke-test-project-building-pipeline-specification

ckairen replied:
So the other parts have been dealt with, and I'll need your help on steps 2,3,6 specified under the "Smoke-Test-Project Building Pipeline Specification" section. The BuildSmokeTestProject is basically the file that does all the steps in the Smoke-Test-Project Building Pipeline Specification" section.

ckairen replied:
the steps in BuildSmokeTestProject.yml presented there right now are just to let the language team devs understand how the PackageInfo can be accessed and see the structure of the pipeline's directory as some of them aren't super familiar with the eng sys pipelines

jsquire replied:
Gotcha.  2 is the part that I'm looking for and was under the assumption was covered here.    We'll need to replace lines 13 - 27 for the below with the package references that you're generating.

jsquire replied:
Once we've got that working, I can integrate into my code updates and we can test end-to-end.

jsquire replied:
I've updated the [Update-Dependencies.ps1](https://github.com/Azure/azure-sdk-for-net/blob/main/common/SmokeTests/SmokeTest/Update-Dependencies.ps1) script to read in the JSON emitted here and rewrite the smoke test project references.

weshaggard replied:
@jsquire just to catch you up on some offline discussion that we had I think Update-Dependencies will end up calling `Get-AllPkgProperties` directly to find all the packages we need to add references to.

jsquire replied:
So, we'd be calling to get the properties in real-time instead of reading the JSON files and would do the T2 filtering as part of that?   Seems reasonable to me.

weshaggard replied:
Correct you would call the in real time and do whatever filter that makes sense. If you just want to limit to track to then $package.IsNewSDK should do that.

weshaggard commented on eng/pipelines/templates/jobs/new.smoke.tests.yml:
```diff
@@ -0,0 +1,42 @@
+jobs:
+  - job: SmokeTest
+    steps:
+      - ${{ if eq(parameters.Daily, true) }}:
```
Why write these to disk as opposed to just read the values directly?

benbp replied:
I think as a boundary between steps/processes it makes local development a lot easier to pass things around via file. You can inspect the output more easily when writing your own tools, and also modify it to test various cases, without needing to re-run the package properties script and/or worry about variables persisting across shell sessions.

weshaggard replied:
We are already running PS code why not just call the Get-AllPkgProperties directly and loop over those instead of going through the process of writing them to disk and then reading them from disk again.

benbp replied:
The step that belongs to the language teams won't necessarily be powershell. I think the pipeline config would be simpler to have a common set of steps and then a simple drop-in yaml step for the language smoke test binary, without requiring extra script handling for passing data to the command line? Otherwise it's going to be harder to move this yaml into eng/common.

weshaggard commented on eng/scripts/smoketest/Save-Package-Properties.ps1:
```diff
@@ -0,0 +1,122 @@
+<#
```
Why are we duplicating this script? What is different from the common one?

ckairen replied:
This script only picks out the track 2 packages, the original one grabs aeverything

jsquire replied:
I'm not sure this is true; I ran this locally and am seeing the T1 packages from the `eventhub` service directory in the output.

weshaggard replied:
Ignoring the track2 vs not logic we should definitely not fully duplicate the script to add that level of filtering. 

Looks like it should filter down to new sdk's https://github.com/Azure/azure-sdk-for-net/pull/25673/files#diff-0b196dd028dd0152730b5a60328c42945f7371f0301f8ff5a4e1b01f90edcf97R98.

weshaggard commented on eng/pipelines/templates/jobs/new.smoke.tests.yml:
```diff
@@ -0,0 +1,42 @@
+          parameters:
+            PackageInfoDir: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/PackageInfo
+
+      - template: /eng/common/TestResources/remove-test-resources.yml
```
Do we need to remove the resources? I wouldn't expect us to even deploy any.

ckairen replied:
No we don't I'll remove this.

ckairen committed:
Package versioning

ckairen committed:
Package versioning

ckairen committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into albertcheng/smoke

ckairen committed:
Adding TimeOut

ckairen committed:
Adding TimeOut

ckairen committed:
Continue on error

ckairen committed:
Continue on error

jsquire commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,63 @@
+  }
+}
+
+return $SmokeTestPackageVersions
```
Why not return the same structure as `Get-AllPkgProperties` and save the need to make multiple calls to query what packages are available?   

From the perspective of the smoke tests script, it's purpose is to rewrite the smoke test project's references.   It would be preferable to have a simple "Get me all the package information for smoke tests" call rather than having to stitch things together.

benbp commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,63 @@
+  -Source $NugetSource `
+  -AllowPrereleaseVersions `
+  -MinimumVersion "$azureCoreVerBase-alpha.202106" `
+  -MaximumVersion "$azureCoreVerBase-alphab" `
```
How does this work?

benbp commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,63 @@
+$azureCoreVerExtension = $azureCorePkg.Version.Replace($azureCoreVerBase, "")
+$azureCoreVerDateStr = $azureCoreVerExtension.split(".")[1]
+$azureCoreVerDate = [DateTime]::ParseExact($azureCoreVerDateStr, "yyyyMMdd", $null)
+$pkgMinVer = $azureCoreVerDate.addMonths(-1).ToString("yyyyMMdd")
```
The code is pretty clear, but I think it's still worth a comment explaining what you're doing here.

benbp commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,63 @@
+    $SmokeTestPackageVersions[$pkgInfo.Name] = $pkgInfo.Version
+  }
+  else {
+    Write-Host "Cannot find alpha version of package $($pkg.Name) after $pkgMinver and before $azureCoreVerDateStr"
+    Write-Host "This may be due to the package being stale or unpublished"
```
```suggestion
    Write-Warning "Cannot find alpha version of package $($pkg.Name) after $pkgMinver and before $azureCoreVerDateStr"
    Write-Warning "This may be due to the package being stale or unpublished"
```

benbp commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,63 @@
+    }
+    else
+    {
+      Write-Host "Did not find any matching package $($pkg.Name)"
+      Write-Host "This error may be due to package not being published"
```
```suggestion
      Write-Warning "Did not find any matching package $($pkg.Name)"
      Write-Warning "This error may be due to package not being published"
```

ckairen committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into albertcheng/smoke

ckairen committed:
Formatting

ckairen committed:
Formatting

ckairen committed:
Added warnings and dynamic azure core min version

ckairen committed:
Added warnings and dynamic azure core min version

jsquire commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,73 @@
+$currentDate = [DateTime]::ParseExact($currentDate, "yyyyMM", $null)
+$azureCoreMinDate = $currentDate.addMonths(-6).ToString("yyyyMM")\
+
+$azureCorePkg = Find-Package -Name 'Azure.Core' `
```
I don't think this covers all of the scenarios that can cause cross-threaded versioning issues.   Any projects that contain project references to another library in the repository would suffer from the same invalid version.   For example,  `Azure.Messaging.EventHubs.Processor` using a project reference to `Azure.Messaging.EventHubs`  or  `Azure.Data.SchemaRegistry` using project references to `Azure.Core.Experimental` and to `Azure.Messaging.EventHubs`.   

I believe that we'll need to do this for all packages.

weshaggard replied:
@jsquire I agree with you that this doesn't cover every scenario but I believe it does handle most of the scenarios and if needed we can harden it further by adding more levels of versioning checks with more than just Core.

One thing to keep in mind is that any Project dependencies within a given service directory (i.e. Azure.Messaging.EventHubs.Process and Azure.Messaging.EventHubs) shouldn't ever be a real issue because they build and publish together. If one of those builds fails than neither of the daily packages will publish so they should never be out of sync. The only time project references might cause things to be out of sync is when they are across service directory boundaries. Azure.Core and Azure.Core.Experimental are probably the two most common cross service directory project references and they publish together from the core service directory so they should also always be in sync with each other as well. So, using Azure.Core as the base dependency I believe should make most scenarios work and if needed we can make adjustments if we find otherwise.

jsquire replied:
@weshaggard:  My fear is that we'll see the same pattern of intermittent failures that we have in the past, and the smoke tests will start being ignored and neglected again.    While I agree that Core is the 80% case, I don't think other scenarios are rare enough that we won't see problems at least a couple of time a month if we restrict this to Core only.

Would it be possible for us to walk the dev feed and validate that all dependency references for a package exist?   If they do not, we step back to the previous version and validate that?  We use the one that is healthy for the smoke tests?   Unless I'm not thinking of it correctly, I believe that would work and we'd have a worst case where the smoke tests were validating some packages a day late.  If both current and previous are broken, we can blow up and raise alarms that there's an issue with the references and someone should investigate.

Thoughts?

weshaggard replied:
Yes I think it would be possible to do a full dependency walk but it would be a lot more complex, and potentially error prone and that is why I think starting with Core is a simpler strategy. If that turns out to still cause issues then we can continue to work on improving the strategy here.

Another thing I worry about if we blindly use the dependency walk is that some packages that might have been failing for a while may end up in pretty stale state, as in we haven't validated them (and any connected packages) for weeks or more and we wouldn't have any indication that they no longer work. I don't want one bad apple to ruin all the other tests. 

Perhaps another middle ground is to add some additional logging here for any cases where we find package versions that are older than the core reference version as those could potentially cause errors. That would at least give us some help when diagnosing any potential failures.

I will also add that I think a lot of the troubles with the older smoke-test pipeline was more around the tests we were running and hopefully that is less of an issue with the new approach.

jsquire replied:
We definitely need to include all `Azure.Core.*` packages for the special logic.    I know of at least 3 different project references to `Azure.Core.Experimental` at the moment.

heaths replied:
Should we also include the track 2s that are still Microsoft.* The ones I know about are all in sdk/core and sdk/extensions. This was done to avoid confusion that these were client libraries, but they are still utilities for working with T2 libraries.

weshaggard commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,73 @@
+
+$currentDate = Get-Date -Format "yyyyMM"
+$currentDate = [DateTime]::ParseExact($currentDate, "yyyyMM", $null)
+$azureCoreMinDate = $currentDate.addMonths(-6).ToString("yyyyMM")\
```
```suggestion
$azureCoreMinDate = $currentDate.addMonths(-1).ToString("yyyyMM")
```
Do we really want 6 months? I would think one month should be enough.

weshaggard commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,73 @@
+$azureCoreVer.IsPreRelease = $false
+$azureCoreVerBase = $azureCoreVer.ToString()
+
+$currentDate = Get-Date -Format "yyyyMM"
```
Why wouldn't this just be `(Get-Date).addMonths(-1).ToString("yyyyMM")`?

weshaggard commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,73 @@
+
+Write-Host "Azure.Core Version: $($azureCorePkg.Version)"
+$azureCoreVerExtension = $azureCorePkg.Version.Replace($azureCoreVerBase, "")
+$azureCoreVerDateStr = $azureCoreVerExtension.split(".")[1]
```
Just to better understand it might be worth adding a comment about what azureCoreVerExtension is expected to look like. I believe it would be something like `-alpha.20211201.01` but worth calling out.

weshaggard commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,73 @@
+
+Write-Host "Smoke Test Package Info:"
+Write-Host $SmokeTestPackageInfo
+return $SmokeTestPackageInfo
```
What do you plan to do with the return here? Is the expectation that Update-Dependencies.ps1 will call this script directly?

ckairen committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into albertcheng/smoke

ckairen committed:
Adding in Get-Package-Version functionality for release (non-Daily)

ckairen committed:
Adding in Get-Package-Version functionality for release (non-Daily)

ckairen commented on eng/pipelines/templates/jobs/new.smoke.tests.yml:
```diff
@@ -0,0 +1,176 @@
+parameters:
```
similar to the old smoke.tests.yml, I put it up to test the changes/integration with Jesse's code. Won't be in the final version when merging. (main difference is the parameters passed to Update-Dependencies.ps1)

ckairen commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,83 @@
+param(
+    [string]$ArtifactsPath
```
This is only used for the release pipeline

jsquire commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,83 @@
+. (Join-Path $PSScriptRoot ../../common/scripts/common.ps1)
+$ErrorActionPreference = 'Continue'
+
+$NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v2"
```
Shouldn't this be:  
https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json

That's what we advertise in the [artifacts feed](https://dev.azure.com/azure-sdk/public/_packaging?_a=connect&feed=azure-sdk-for-net).

weshaggard replied:
For some reason the PS package cmdlets don't work with the v3 nuget endpoint they require the v2 endpoint.

jsquire replied:
> For some reason the PS package cmdlets don't work with the v3 nuget endpoint they require the v2 endpoint.

The `PackageManagement` module in PowerShell is deprecated and hasn't been in active development for a while.  The PowerShell team recommends that it not be used.   There's some additional context in #26695.

jsquire commented on eng/scripts/smoketest/Get-Package-Version.ps1:
```diff
@@ -0,0 +1,83 @@
+param(
```
This script appears to be intended to be called from a command line and not really imported/called as part of another script.

Can we consider wrapping things into a function to make calling it more explicit?  Something like:

```powershell
param(
    [string]$ArtifactsPath
)
. (Join-Path $PSScriptRoot ../../common/scripts/common.ps1)
$ErrorActionPreference = 'Continue'

function Get-SmokeTestPkgProperties
{
    Param
    (
        [string]$ArtifactsPath
    )

    // ... CODE ..
}

if ($ArtifactsPath -ne $null)
{
    Get-SmokeTestPkgProperties($ArtifactsPath)
}
```

jsquire replied:
Could we also please disable writing to host by default and add a param to enable it?   The output from calling this overwhelms the output from callers and makes it quite difficult to follow the flow.

weshaggard replied:
We can use Write-Verbose so we can turn it on only for debugging as necessary.

heaths replied:
Yes, please use `Write-Verbose` or `Write-Debug` and not invent another message writing system for PowerShell that designed a consistent one.

jsquire reviewed (commented):
I ran a prototype check against the packages returned by this.   I'm seeing quite a few issues with packages unable to be restored while referencing the [dev feed](https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json).   

For example:

```text
Unable to find package Microsoft.Extensions.Azure with version (>= 1.2.0-beta.1)
  - Found 404 version(s) in azure-sdk-for-net [ Nearest version: 1.2.0-alpha.20220223.1 ]
  - Found 9 version(s) in nuget.org [ Nearest version: 1.1.1 ]
  - Found 0 version(s) in azure-sdk-tools
```

I'm happy to provide the set of references returned, if that's helpful.

ckairen committed:
Merge branch 'eng/smoke-tests' of https://github.com/jsquire/azure-sdk-for-net into albertcheng/smoke

ckairen committed:
pipeline test

ckairen committed:
pipeline test

ckairen committed:
add install dotnet in smoke test

ckairen committed:
add install dotnet in smoke test

heaths commented on common/SmokeTests/SmokeTest/Program.cs:
```diff
@@ -1,37 +1,227 @@
+{
+    Console.ForegroundColor = color ?? ConsoleColor.Red;
+    Console.Error.WriteLine(message);
+    Console.ResetColor();
```
Tip: it's unlikely, but if anything throws this won't work. Instead, I like to use this pattern:

```csharp
var previousForeground = Console.ForegroundColor;
Console.ForegroundColor = color ?? ConsoleColor.Red;
try
{
  // Do whatever
}
finally
{
  Console.ForegroundColor = previousForeground;
}
```

Granted, I wouldn't expect just writing to the console to fail, but if this ever does anything more - like maybe formatting an event, exception, etc. - it's more likely.

jsquire replied:
I don't see this doing anything additional, but the extra safety won't hurt.    Updated patterns in the main PR.

weshaggard committed:
Test ignoring NU1605

weshaggard committed:
Exclude some references on unsupported platforms

heaths commented on common/SmokeTests/SmokeTest/SmokeTest.csproj:
```diff
@@ -37,4 +37,9 @@
   </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'netcoreapp3.1'">
+    <!-- This package is supports netcoreapp3.1 -->
```
```suggestion
    <!-- This package supports netcoreapp3.1 -->
```

weshaggard commented:
@jsquire could you please help have a look at https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1432298&view=logs&j=f3570516-c8fe-5ce9-92b2-b737df25d5ea&t=8b1263e5-6598-5fef-152d-3338b5972413? Looks like the test code is now hitting load failures but there isn't any good information as to why.

jsquire commented:
> @jsquire could you please help have a look at https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1432298&view=logs&j=f3570516-c8fe-5ce9-92b2-b737df25d5ea&t=8b1263e5-6598-5fef-152d-3338b5972413? Looks like the test code is now hitting load failures but there isn't any good information as to why.

Looks like `Microsoft.Identity.Client.TokenCache` is being detected as a parameterless constructor but is throwing because there are missing parameters.    I'll investigate; likely we missed a corner/special case.

jsquire commented on common/SmokeTests/SmokeTest/Program.cs:
```diff
@@ -1,37 +1,227 @@
+                    Console.WriteLine($"\tConstructed: '{ type.FullName }'");
+                }
+            }
+            catch (TargetInvocationException ex) when (ex.InnerException is NotImplementedException)
+            {
+                // Expected; this occurs when the parameterless constructor is present but has been marked obsolete.  Since we do
+                // not report types without a parameterless constructor, ignore the type and do not log.
+            }
```
```suggestion
            catch (TargetInvocationException ex) when (ex.InnerException is NotImplementedException)
            {
                // Expected; this occurs when the parameterless constructor is present but has been marked obsolete.  Since we do
                // not report types without a parameterless constructor, ignore the type and do not log.
            }
            catch (TargetInvocationException ex) when (ex.InnerException is ArgumentException)
            {
                // Expected; this occurs when the parameterless constructor is present but the type cannot be constructed due to failing
                // an internal state validation.  Since we do not report types without a parameterless constructor, ignore the type
                // and do not log.
            }
```

jsquire committed:
Fix constructor failure for TokenCache

Adding an additional corner-case guard for types that have a parameterless constructor but fail an internal state check.

AlexGhiondea commented on common/SmokeTests/SmokeTest/Program.cs:
```diff
@@ -1,37 +1,233 @@
     }
+    catch (Exception ex)
+    {
+        ReportError();
+        LogError($"{Environment.NewLine }[{ assembly.FullName }] could not be fully processed.  Error: { ex }{ Environment.NewLine }");
```
Do we want to flag all the types with a problem in an assembly instead of stopping at the first failure?

jsquire replied:
We don't stop on failure;  we write output about what type failed and keep going until we reach the end.     That said, we don't collect failures and do a nice "this is where the problems were"  - so that may be an improvement to the "go look through the output for errors" experience that we've got today.

jsquire committed:
Additional edge cases for type/assembly loading.

jsquire committed:
Additional "Not Found" corner cases

Handling additional "not found" cases for platform-specific assemblies during load or type activation.

jsquire committed:
.NET Framework corner cases

Additional error handling for .NET Framework corner cases around attempting to load unsupported assembly types, such as UWP assemblies.

jsquire committed:
.NET Framework activation guards for COM types

Additional guards for COM-based type activation scenarios for the .NET Framework target.

jsquire committed:
.NET Framework fix for COM Error detection

jsquire committed:
Additional context for ReflectionTypeLoadException

jsquire committed:
Test run without recursive assembly discovery

jsquire committed:
Removing recursive assembly discovery

I'm not convinced that the recursive assembly discovery is providing value; assemblies needed to reflect over types are being dynamically loaded on inspection.

jsquire committed:
Fixing comment to agree with the code.

jsquire committed:
Fixing logic error

weshaggard committed:
Update package version selection strategy

The verison selection process works as following:

Find at least one day old Azure.Core version so that other
packages have a chance to have build between the time core
and smoke test pipeline run.

Find other alpha packages that match exactly the same day
and if a package doesn't exist that matches the day then
we exclude it from the smoke testing for that day.

This will definitely lead to various package combinations but
they should all agree on version at least which will help with
the versioning conflicts.

weshaggard commented on eng/pipelines/templates/jobs/new.smoke.tests.yml:
```diff
@@ -0,0 +1,179 @@
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          ArmTemplateParameters: $(azureCloudArmParameters)
+          Location: "westus"
+        Windows_NetCoreApp (AzureUSGovernment):
```
We will want to clean-up these matrix entries because we are not deploying live test resources any longer so we don't need to test in different clouds.

weshaggard commented on common/SmokeTests/SmokeTest/Program.cs:
```diff
@@ -1,37 +1,278 @@
+
+void ReportError() => returnCode = -1;
+
+void LogError(string message, ConsoleColor? color = default)
```
We should add an option that will write this error in the magic DevOps way so they will be prompted to the error list in the pipeline to allow for easier finding. See https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/logging.ps1#L19 where we do this in PS. All we need to do is check for the SYSTEM_TEAMPROJECTID env variable and if it is set format the output with the magic header.

jsquire replied:
I'll work this in.

jsquire committed:
Add VSO formatting to logging

jsquire committed:
Fixing log formatting

Error and Debug log formats were reversed.

jsquire committed:
Client constructor now uses LogInformation

ckairen committed:
pull

ckairen committed:
Matrix cleanup

jsquire commented:
@ckairen:  At this point, this PR is the authoritative copy of the updated Smoke Tests code, as we've all been pushing directly to this branch.    Can you please restore the files that were removed here?

heaths commented on eng/pipelines/templates/jobs/new.smoke.tests.yml:
```diff
@@ -45,83 +45,8 @@ jobs:
-          TestTargetFramework: net461
-          SubscriptionConfiguration: $(sub-config-cn-test-resources)
-          ArmTemplateParameters: $(azureChinaCloudArmParameters)
-          Location: "chinaeast2"
```
Some libraries actually do different between Framework and Core. I wonder if it's worth keeping these and only these two. Thoughts?

heaths replied:
Though, probably just AzureCloud since propagation of new api-versions to sovereign clouds trails AzurePublic, and some sovereign clouds like Mooncake may not get all of them.

ckairen replied:
We're not doing live tests anymore [ref](https://github.com/Azure/azure-sdk-for-net/pull/25673#discussion_r830390592), so cloud wouldn't matter

jsquire committed:
Update common/SmokeTests/SmokeTest/SmokeTest.csproj

Co-authored-by: Heath Stewart <heaths@microsoft.com>

ckairen committed:
Decrease output noise

ckairen committed:
Merge branch 'albertcheng/smoke' of https://github.com/Azure/azure-sdk-for-net into albertcheng/smoke

ckairen committed:
Adding in min matrix

ckairen commented:
@weshaggard Are there any suggestions that you want to put it out before this PR is merged?

weshaggard commented on common/SmokeTests/smoke-tests.yml:
```diff
@@ -2,7 +2,7 @@ variables:
 
 jobs:
-  - template: /eng/pipelines/templates/jobs/smoke.tests.yml
+  - template: /eng/pipelines/templates/jobs/new.smoke.tests.yml
```
Did you want to fix this to just be smoke.tests.yml before this merge or did you plan to clean that up afterwards?

ckairen replied:
This will be changed in follow-up PR

weshaggard reviewed (approved):
Left one comment about new.smoke.tests.yml and when plan to transition to smoke.tests.yml.

I had a quick look over the latest smoke-test run and I saw a ton of warnings which might end up being very noisy when trying to figure out any potential issues. We should have a closer look at those once we get these changes in to see if there is anything we can do to help reduce that noise.

ckairen commented:
/check-enforcer override
